projects/mci/test/test_drive_fsm.c

```diff
@@ -0,0 +1,151 @@
+#include "can.h"
+#include "can_ack.h"
+#include "can_msg_defs.h"
+#include "can_transmit.h"
+#include "delay.h"
+#include "event_queue.h"
+#include "exported_enums.h"
+#include "gpio.h"
+#include "interrupt.h"
+#include "log.h"
+#include "ms_test_helpers.h"
+#include "soft_timer.h"
+#include "test_helpers.h"
+
+#include "drive_fsm.h"
+#include "mci_events.h"
+#include "motor_controller.h"
+
+static CanStorage s_can_storage;
+static MotorControllerStorage s_mci_storage;
+
+void prv_setup_system_can() {
+  CanSettings can_settings = {
+    .device_id = SYSTEM_CAN_DEVICE_MOTOR_CONTROLLER,
+    .bitrate = CAN_HW_BITRATE_500KBPS,
+    .rx_event = MCI_CAN_EVENT_RX,
+    .tx_event = MCI_CAN_EVENT_TX,
+    .fault_event = MCI_CAN_EVENT_FAULT,
+    .tx = { GPIO_PORT_A, 12 },
+    .rx = { GPIO_PORT_A, 11 },
+    .loopback = true,
+  };
+
+  can_init(&s_can_storage, &can_settings);
+}
+
+void prv_mci_storage_init(void *context) {
+  MotorControllerStorage *storage = context;
+  PrechargeStorage precharge_storage = { .precharge_control = { .port = GPIO_PORT_A, .pin = 9 },
+                                         .precharge_control2 = { .port = GPIO_PORT_B, .pin = 1 },
+                                         .precharge_monitor = { .port = GPIO_PORT_B, .pin = 0 },
+                                         .state = MCI_PRECHARGE_DISCHARGED };
+  storage->precharge_storage = precharge_storage;
+}
+
+void setup_test(void) {
+  event_queue_init();
+  interrupt_init();
+  soft_timer_init();
+  prv_setup_system_can();
+  prv_mci_storage_init(&s_mci_storage);
+  precharge_control_init(&s_mci_storage);
+  TEST_ASSERT_TRUE(drive_fsm_init(&s_mci_storage) == STATUS_CODE_OK);
+}
+
+void teardown_test(void) {}
+
+static StatusCode prv_ack_callback(CanMessageId msg_id, uint16_t device, CanAckStatus status,
+                                   uint16_t num_remaining, void *context) {
+  CanAckStatus *expected_status = context;
+  TEST_ASSERT_TRUE(*expected_status == status);
+  return STATUS_CODE_OK;
+}
+
+void test_neutral(void) {
```

Comment:
bad name, you're testing transition from neutral to drive then to reverse...

Reply:
Nah the test file tests states (and all exiting transitions) rather than a test function for each transition